frame/election-provider-support/solution-type/src/single_page.rs

```diff
@@ -134,6 +134,12 @@ pub(crate) fn generate(def: crate::SolutionDef) -> Result<TokenStream2> {
 						}
 					}
 				};
+
+				// Make sure that the voter bound is binding.
+				use frame_support::traits::Get;
+				if #struct_name.voter_count() as u32 > #max_voters::get() {
```

Comment:
while this is correct, it needs to re-iterate the whole struct, which we do just above (in the guts of `#from_impl`). Alternatively, you can try and populate a counter while we iterate above.

Reply:
`assignments.len()` actually represents the number of voters, so I could use that instead?

Reply:
Yes, just do it before doing any other check, so we can early exist if this is going to fail.